### PR TITLE
Refactor preprocess features

### PR DESCRIPTION
**Pull request by roshankern**
This PR is ready for review!

In this PR, the preprocess-features module is refactored to include both merging of the CP/DP features and normalization of all of these features.

### CODE REVIEW

File: 3.preprocess-features/3a.merge-features/3a.merge-cp-dp-features.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+# ### Merge Features
+# 
+
+# In[3]:
+
+
+for cp_output_path in cp_features_save_path.iterdir():
+    plate = cp_output_path.name
+    print(f"Merging plate {plate} ...")
+
```
Comment (axiomcura) [inline 3.preprocess-features/3a.merge-features/3a.merge-cp-dp-features.py]: consider provinding a general comment explaining what the for loop is doing just for readers to know what it is going to happen.

File: 3.preprocess-features/3a.merge-features/merge-utils.py [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+    str
+        well string in DP format
+    """
+
+    row = int(cp_well[1:3])
+    formatted_row = chr(64 + row)
+
+    col = int(cp_well[4:6])
+
+    return f"{formatted_row}{col}"
```
Comment (axiomcura) [inline 3.preprocess-features/3a.merge-features/merge-utils.py]: consider providing some inline comments on how the naming scheme is generated.

File: 3.preprocess-features/3a.merge-features/merge-utils.py [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+    merge CP and DP single-cell data from the same image (plate, well, site combination)
+
+    Parameters
+    ----------
+    cp_image_data : pd.DataFrame
+        cp single-cell data from the image to merge cells from
+    dp_image_data : pd.DataFrame
+        dp single-cell data from the image to merge cells from
+    add_cell_uuid : bool, optional
+        whether or not single-cell UUIDs should be added (useful for identifying a particular cell), by default True
```
Comment (axiomcura) [inline 3.preprocess-features/3a.merge-features/merge-utils.py]:
```suggestion
        whether or not single-cell UUIDs should be added (useful for identifying a particular cell), by default False
```
Consider revising docstring. The `add_cell_uuid` is set to `False` by default in the function parameters.

File: 3.preprocess-features/3b.normalize-features/normalization-utils.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    Returns
+    -------
+    StandardScaler
+        normalization scaler for merged feature cells
+    """
+
+    # find all cells that have had no reagent applied
+    negative_control_single_cells = plate_merged_single_cells.loc[
+        plate_merged_single_cells["Metadata_Reagent"] == "ARID1B-2"
+    ]  # CHANGE TO "no-reagent"
```
Comment (axiomcura) [inline 3.preprocess-features/3b.normalize-features/normalization-utils.py]: Is this comment supposed to be here? if not, considered removing it.  If it is, consider moving it to the top of the code block.

File: 3.preprocess-features/3b.normalize-features/3b.normalize-merged-features.py [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+scaler_save_dir = pathlib.Path("normalization-scalers/")
+scaler_save_dir.mkdir(parents=True, exist_ok=True)
+
+
+# ### Normalize merged single-cell data
+# 
+
+# In[3]:
+
+
```
Comment (axiomcura) [inline 3.preprocess-features/3b.normalize-features/3b.normalize-merged-features.py:53]: consider providing a single comment providing a general explanations of what is occurring in this for loop.

**Review by axiomcura — APPROVED** [review]
LGTM left some suggestions 

Feel free to ask me anything!

File: 3.preprocess-features/3b.normalize-features/normalization-utils.py (hunk 4 above)
Comment (roshankern) [reply 3.preprocess-features/3b.normalize-features/normalization-utils.py]: I needed to implement that comment, thanks for pointing it out!

**Pull request merged**